packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -192,23 +189,35 @@ contract OptimisticGovernor is Module, Lockable {
         // proposalHashes[id] = keccak256(abi.encodePacked(proposalData));
         proposalHashes[id] = keccak256(abi.encode(_transactions));
 
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/OptimisticOracleInterface.sol
+
+        // optimisticOracle.requestAndProposePriceFor(
+        //     identifier,
+        //     uint32(time),
+        //     ancillaryData,
+        //     collateral,
+        //     0,
+        //     bond,
+        //     uint256(liveness),
+        //     address(this),
+        //     // Canonical value representing "True"; i.e. the transactions are valid.
+        //     int256(1e18)
+        // );
+        optimisticOracle.requestPrice(identifier, uint32(time), ancillaryData, collateral, 0);
+
+        uint256 totalBond = optimisticOracle.setBond(identifier, uint32(time), ancillaryData, bond);
+
         // Get the bond from the proposer and approve the bond and final fee to be used by the oracle.
-        uint256 totalBond = finalFee + bond;
         collateral.safeTransferFrom(msg.sender, address(this), totalBond);
-        collateral.safeIncreaseAllowance(address(skinnyOptimisticOracle), totalBond);
+        collateral.safeIncreaseAllowance(address(optimisticOracle), totalBond);
 
-        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
-        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
-        skinnyOptimisticOracle.requestAndProposePriceFor(
+        optimisticOracle.proposePriceFor(
+            address(this),
```

Comment:
I think the proposer should be `msg.sender`, not `this`. This is because we want the returned bond and any rewards to be paid out to them, not this contract.